server/api.go

```diff
@@ -95,13 +101,34 @@ func (p *Plugin) connectUserToGitHub(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	conf := p.getOAuthConfig()
+	privateAllowed := false
+	pVal := r.URL.Query().Get("private")
+	if pVal == "ok" {
+		privateAllowed = true
+	}
 
-	state := fmt.Sprintf("%v_%v", model.NewId()[0:15], userID)
+	conf := p.getOAuthConfig(privateAllowed)
 
-	p.API.KVSet(state, []byte(state))
+	state := OAuthState{
+		UserID:         userID,
+		Token:          model.NewId()[:15],
+		PrivateAllowed: privateAllowed,
+	}
 
-	url := conf.AuthCodeURL(state, oauth2.AccessTypeOffline)
+	stateBytes, err := json.Marshal(state)
+	if err != nil {
+		http.Error(w, "json marshal failed", http.StatusInternalServerError)
+		return
+	}
+
+	appErr := p.API.KVSet(state.Token, stateBytes)
+	if appErr != nil {
+		fmt.Println(appErr.Error())
```

Comment:
Intentional to log error to STDOUT? (here, and existing cases below)

Reply:
I removed this one, since it's in response to the user's click. The ones below are in response to the OAuth flow, and so any returned http error will disappear and never be surfaced for a sysadmin. That's why I'm assuming they're intentional. :)